src/api/handlers.rs

```diff
@@ -0,0 +1,253 @@
+        resource=tracing::field::Empty,
+        allowed=tracing::field::Empty,
+        mutated=tracing::field::Empty,
+        response_code=tracing::field::Empty,
+        response_message=tracing::field::Empty,
+    ),
+    skip_all)]
+///  Run a validation in "audit" mode.
+pub(crate) async fn audit_handler(
+    extract::State(state): extract::State<Arc<ApiServerState>>,
+    extract::Path(policy_id): extract::Path<String>,
+    extract::Json(admission_review): extract::Json<AdmissionReviewRequest>,
+) -> Result<Json<AdmissionReviewResponse>, (StatusCode, ApiError)> {
+    populate_span_with_admission_request_data(&admission_review.request);
+
+    let response = acquire_semaphore_and_evaluate(
+        state,
+        policy_id,
+        ValidateRequest::AdmissionRequest(admission_review.request),
+        RequestOrigin::Audit,
+    )
+    .await
+    .map_err(handle_evaluation_error)?;
+
+    populate_span_with_policy_evaluation_results(&response);
+
+    Ok(Json(AdmissionReviewResponse::new(response)))
+}
+
+// note about tracing: we are manually adding the `policy_id` field
+// because otherwise the automatic "export" would cause the string to be
+// double quoted. This would make searching by tag inside of Jaeger ugly.
+// A concrete example: the automatic generation leads to the creation
+// of `policy_id = "\"psp-capabilities\""` instead of `policy_id = "psp-capabilities"`
+#[tracing::instrument(
+    name = "validation",
+    fields(
+        request_uid=tracing::field::Empty,
+        host=crate::config::HOSTNAME.as_str(),
+        policy_id=policy_id.as_str(),
+        name=tracing::field::Empty,
+        namespace=tracing::field::Empty,
+        operation=tracing::field::Empty,
+        subresource=tracing::field::Empty,
+        kind_group=tracing::field::Empty,
+        kind_version=tracing::field::Empty,
+        kind=tracing::field::Empty,
+        resource_group=tracing::field::Empty,
+        resource_version=tracing::field::Empty,
+        resource=tracing::field::Empty,
+        allowed=tracing::field::Empty,
+        mutated=tracing::field::Empty,
+        response_code=tracing::field::Empty,
+        response_message=tracing::field::Empty,
+    ),
+    skip_all)]
+/// Validate a request against a policy.
+pub(crate) async fn validate_handler(
+    extract::State(state): extract::State<Arc<ApiServerState>>,
+    extract::Path(policy_id): extract::Path<String>,
+    JsonExtractor(admission_review): JsonExtractor<AdmissionReviewRequest>,
+) -> Result<Json<AdmissionReviewResponse>, (StatusCode, ApiError)> {
+    populate_span_with_admission_request_data(&admission_review.request);
+
+    let response = acquire_semaphore_and_evaluate(
+        state,
+        policy_id,
+        ValidateRequest::AdmissionRequest(admission_review.request),
+        RequestOrigin::Validate,
+    )
+    .await
+    .map_err(handle_evaluation_error)?;
+
+    populate_span_with_policy_evaluation_results(&response);
```

Comment:
As far as I can see, the tracing for policies in monitor mode will always have result as "allowed". Which is not desired. The metrics should show the original evaluation result. Otherwise, the users will not get the value of the monitor mode which is see the original results before moving the policy to protect mode.

Reply:
I Agree with this. Still this isn't a regression from this PR, so I'm ok opening an issue and tackling it later. I suppose adding a `mode` field and giving value to `allowed` correctly will suffice.

Reply:
I think we should  create a new issue for that, and tackle that outside of this PR.

I also prefer to keep the current behavior, but extend the trace to have:

- a new field `more` that states whether the policy is operating in `monitor` or `protect` mode
- a new field `raw_result` (or something else) that contains the boolean value of the evaluation result **before** the monitor mode changes that. We could have this field added to all the policies, regardless of their operating mode, or we could have it added only to the traces emitted by policies operating in `monitor` mode

At the end of the day, I want an operator to be able to run a Jaeger query like: `select * from traces where operation_mode = "monitor" and raw_result = false`